zkvm/src/vm.rs

```diff
@@ -51,6 +56,99 @@ pub struct Tx {
     pub proof: R1CSProof,
 }
 
+impl Tx {
+    /// Serializes the tx into a byte array.
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut buf = Vec::with_capacity(self.serialized_size());
+        encoding::write_u64(self.header.version, &mut buf);
+        encoding::write_u64(self.header.mintime, &mut buf);
+        encoding::write_u64(self.header.maxtime, &mut buf);
+
+        encoding::write_size(self.program.len(), &mut buf);
+        buf.extend(&self.program);
+        buf.extend_from_slice(&self.signature.to_bytes());
+        buf.extend_from_slice(&self.proof.to_bytes());
+        buf
+    }
+
+    /// Returns the size in bytes required to serialize the `Tx`.
+    pub fn serialized_size(&self) -> usize {
+        // header is 8 bytes * 3 fields = 24 bytes
+        // program length is 4 bytes
+        // program is self.program.len() bytes
+        // signature is 64 bytes
+        // proof is 14*32 + the ipp bytes
+        8 * 3 + 4 + self.program.len() + 64 + self.proof.serialized_size()
```

Comment:
```suggestion
        self.header.serialized_size() + 4 + self.program.len() + 64 + self.proof.serialized_size()
```

maybe factor `8*3` into the TxHeader's method?